lib/authorization/assertion/twitter_auth.rb

```diff
@@ -0,0 +1,49 @@
+module Authorization
+  module Assertion
+    class TwitterAuth
+      def initialize(access_token, access_token_secret)
+        @access_token = access_token
+        @access_token_secret = access_token_secret
+        @twitter_client = twitter_client
+        @user_data = user_data
+      end
+
+      def user_data
+        @twitter_client.user
+      end
+
+      def image
+        img = URI.parse(@user_data.profile_image_url)
+        Net::HTTP.get_response(img)
+      end
+
+      def user!
+        User.where(twitter_id: @user_data.id).first if @user_data.present?
+      end
+
+      def import_friends
+        return unless @user_data.present?
+        user = user!
```

Comment:
You don't need to save `user!`, you can use it directly on line 30 as `follower: user!,`